governance/writing-style.md

```diff
@@ -0,0 +1,66 @@
+
+### Fixing QA errors
+
+Every tool provides quite descriptive messages regarding the location of the error and how it should be fixed, but below is a quick guide for additional help and reference.
+
+1. **Markdown linting.** ```markdownlint-cli2``` tool is configured to automatically fix those issues that it can fix automatically, but if there are issues that can't be fixed automatically, an error with file and location will be printed out for your reference. For example, ```writing-style.md MD024/no-duplicate-heading/no-duplicate-header: Multiple headings with the same content markdownlint(MD024) [Ln 54, Col 1]``` is an error message that states that it has identified a heading of the same level on **line 54** of **writing-style.md** file, which is where you would need to update it to resolve the issue. More rules with explanations can be found [here](https://github.com/DavidAnson/markdownlint/blob/main/README.md#rules--aliases).
+
+2. **Spelling.** ```cSpell``` tool has a configuration file called [cspell.json](./cspell.json), that is located in the root of the repo, and it defines a set of rules that the tool following during its execution. It checks the words towards a set of default dictionaries, words that we have identified to always be correct and a set of patterns that the tool shall ignore and not scan (like code blocks and Markdown/HTML specific link syntax, etc.). In addition we have created a custom TAG ENV dictionary, [tag-custom-dictionary-cspell.txt](./tag-custom-dictionary-cspell.txt), where we can add words that are correct, but are not present in default dictionaries. In case you get ```Unknown word``` error you will need to verify that the word is correctly spelled and if not, update it accordingly on the line and in the file that the tool is referring to. If the word is correct, you may add it to the TAG ENV custom dictionary mentioned above. There are common words that may not exist in known dictionaries and may be challenging to detect with filtering rules, like names, which may still be flagged by the tool. It's not needed to add this type of words to our custom TAG ENV dictionary, the easiest would be to add an ignore flag. You can add an ignore flag by defining which word to ignore: ```<!-- cSpell:ignore WORD_TO_IGNORE -->```, or by ignoring the whole next line where the word(s) are: ```<!-- cspell:disable-next-line -->```, or by ignoring a longer content block by setting ```<!-- cSpell:disable -->``` before start of the block to ignore and ```<!-- cSpell:enable -->``` after the end of the block to ignore. For additional details, please check [cSpell documentation](https://cspell.org/configuration/document-settings).
```

Comment:
```suggestion
2. **Spelling.** ```cSpell``` tool has a configuration file called [cspell.json](../cspell.json), that is located in the root of the repo, and it defines a set of rules that the tool following during its execution. It checks the words towards a set of default dictionaries, words that we have identified to always be correct and a set of patterns that the tool shall ignore and not scan (like code blocks and Markdown/HTML specific link syntax, etc.). In addition we have created a custom TAG ENV dictionary, [tag-custom-dictionary-cspell.txt](../tag-custom-dictionary-cspell.txt), where we can add words that are correct, but are not present in default dictionaries. In case you get ```Unknown word``` error you will need to verify that the word is correctly spelled and if not, update it accordingly on the line and in the file that the tool is referring to. If the word is correct, you may add it to the TAG ENV custom dictionary mentioned above. There are common words that may not exist in known dictionaries and may be challenging to detect with filtering rules, like names, which may still be flagged by the tool. It's not needed to add this type of words to our custom TAG ENV dictionary, the easiest would be to add an ignore flag. You can add an ignore flag by defining which word to ignore: ```<!-- cSpell:ignore WORD_TO_IGNORE -->```, or by ignoring the whole next line where the word(s) are: ```<!-- cspell:disable-next-line -->```, or by ignoring a longer content block by setting ```<!-- cSpell:disable -->``` before start of the block to ignore and ```<!-- cSpell:enable -->``` after the end of the block to ignore. For additional details, please check [cSpell documentation](https://cspell.org/configuration/document-settings).
```